### ISSUE TITLE
Tiedtke deep/shallow split is CAPE-based, not ECHAM's moisture-convergence test (30x entrainment consequence)

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#699]
Found while porting ECHAM's `cubasmc` mid-level trigger (#697), which made the `ktype` assignment worth reading end-to-end. Parent: #614.

## What's wrong

ECHAM decides deep vs shallow for a `cubase`-initiated plume from the column **moisture budget**, not from CAPE (`mo_cumastr.f90:571-573`):

```fortran
zhelp      = MAX(0._wp, -1.1_wp*pqhfla(jl)*grav)      ! 1.1 x surface latent flux
zktype(jl) = FSEL(zhelp - zdqcv(jl), 2._wp, 1._wp)    ! deep iff zdqcv > zhelp
zentr(jl)  = FSEL(zhelp - zdqcv(jl), entrscv, entrpen)
```

`zdqcv` is the mass-weighted column integral of `pqte` — total moisture convergence. The test is therefore "is there large-scale moisture convergence *beyond* what the surface is evaporating?", which is the physical distinction between a deep convergent regime and a shallow locally-driven one. A second pass then demotes 1 -> 2 if the cloud turns out thinner than 200 hPa (`mo_cumastr.f90:752`).

jcm instead splits on a CAPE sigmoid about 1000 J/kg:

```python
w_deep = jax.nn.sigmoid((cape - 1000.0) / config.smooth_type_j)
```

with no moisture-convergence term and no post-hoc depth demotion. The 1000 J/kg threshold has no ECHAM counterpart.

## Why it matters

The two criteria disagree systematically, and the consequence is not cosmetic: `ktype` selects the base entrainment rate, and `entrpen` (1e-4 /m) and `entrscv` (3e-3 /m) differ by **30x**. A column mislabelled deep entrains 30x too weakly and produces a far too deep, too undilute plume; one mislabelled shallow does the reverse.

The regimes where they part company are exactly the interesting ones: high-CAPE-but-non-convergent columns (a hot, dry, subsiding subtropical continental afternoon has large CAPE and no convergence — ECHAM says shallow, jcm says deep), and low-CAPE-but-strongly-convergent columns (frontal and ITCZ ascent — ECHAM says deep, jcm says shallow).

## What it needs

- **`zdqcv`**: the column integral of the moisture tendency. The pieces are already present — `tiedtke_nordeng_convection` receives `moisture_tend_profile` (the same-step vdiff `pqte` analog) and integrates it over the sub-cloud layer for the `zdqpbl` closure. `zdqcv` is the same integral over the **whole column**, so it is a few lines given what is already threaded.
- **`pqhfla`**: the surface latent flux, already available as `moisture_supply` (the delivered surface evaporation, in kg/m2/s, i.e. `pqhfla*grav` in ECHAM's units up to sign).
- **The depth demotion** (`zpbmpt < 2e4` -> `ktype = 2`, and `zentr = entrscv` with it). jcm computes cloud top dynamically inside `calculate_updraft`, so this is a post-updraft correction to the entrainment rate — structurally the same place the Nordeng CAPE rescale already applies.
- Keep it differentiable: the current split is a sigmoid whose width is a tunable parameter, and the replacement should be too (a sigmoid in `zdqcv - zhelp`).

**Caveat, same as the `zdqpbl` closure**: jcm's `pqte` analog carries the vdiff part but not the large-scale ADVECTIVE part of moisture convergence, because the dycore applies advection after physics. For `zdqpbl` that omission is conservative (a smaller supply). For `zdqcv` it is **not** obviously conservative — advective convergence is precisely the signal the deep/shallow test is looking for, and dropping it biases the test toward shallow in exactly the frontal/ITCZ columns that motivate the change. So this issue probably cannot be closed properly without the same-step advective tendency plumbing, and that dependency should be settled first rather than discovered halfway in.

## Refs

Cross-referenced from the `ktype` comment in `jcm/physics/convection/tiedtke_nordeng/tiedtke_nordeng.py`. Related: #697 (mid-level trigger, which established that `ktype` names the trigger rather than diagnosing the column), #684 (the `cubase` walk), #676 (Tiedtke ledger gaps).
